### PR TITLE
Add number to bmw_connected_drive documentation

### PR DESCRIPTION
**Pull request by rikroe**
## Proposed change

Add documention on the new `number` entities for EV charging in BMW Connected Drive.

~~Needs an additional commit once #26365 is merged (delete one line from there).~~

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/91081
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by rikroe** [comment]
Hey @frenck, as https://github.com/home-assistant/core/pull/91081 got merged into 2023.5 (thanks!), this should also be included.

**Comment by rikroe** [comment]
Also, I was wondering if https://github.com/home-assistant/core/pull/88759 and https://github.com/home-assistant/core/pull/91081 could be seen as noteworthy change, as BMW/Mini EV owners now are able to control vehicle charging from HA.

File: source/_integrations/bmw_connected_drive.markdown [hunk 1]
```diff
@@ -144,9 +147,16 @@ If you have a (PH)EV, you can control the charging process through Home Assistan
 Using these selects will impact the state of your vehicle. Use them with care!
 
 - **Charging Mode**: Vehicle can be set to `IMMEDIATE_CHARGING` (charge as soon as plugged in) or `DELAYED_CHARGING` (charge only if within charging window). It can be used to start/stop charging if the charging window is set accordingly.
-- **Target SoC**: The vehicle will charge until this battery level is reached. Not available on all EVs.
 - **AC Charging Limit**: The maximum current a vehicle will charge with. Not available on all EVs.
 
+## Numbers
+
+If you have a (PH)EV, you can control the charging process through Home Assistant. The number entities are created automatically depending on your vehicle's capabilities and can be changed from the UI or using the `number.set_value` service. Please see the [number documentation](/integrations/number/) for more information.
```
Comment (frenck) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
If you have a (PH)EV, you can control the charging process through Home Assistant. The number entities are created automatically depending on your vehicle's capabilities and can be changed from the UI or using the `number.set_value` service. For more information, please see the [number documentation](/integrations/number/).
```

**Review by frenck — APPROVED** [review]
Thanks, @rikroe 👍 

../Frenck

**Pull request merged**